docs/proposals/201909_thanos_sharding.md

```diff
@@ -0,0 +1,178 @@
+---
+title: Thanos Sharding for Long Term Retention Storage
+type: proposal
+menu: proposals
+status: approved
+owner: bwplotka
+---
+
+### Related Tickets
+
+* https://github.com/thanos-io/thanos/issues/1034 (main)
+* https://github.com/thanos-io/thanos/pull/1245 (attempt on compactor)
+* https://github.com/thanos-io/thanos/pull/1059 (attempt on store)
+* https://github.com/thanos-io/thanos/issues/1318 (subpaths, multitenancy)
+* Store issues because of large bucket: 
+  * https://github.com/thanos-io/thanos/issues/814
+  * https://github.com/thanos-io/thanos/issues/1455
+
+## Summary
+
+This document describes the motivation and design of the sharding of Thanos components in terms of operations against object storage.
+Additionally we touch possibility for smarter pre-filtering of shards on the Querier.
```

Comment:
This sentence feels a bit wordy maybe something like `Additionally we introduce ability to pre filter shards in Thanos Querier`